File: automation-test-playwright/pages/staff/StaffDashboardPage.ts
import { type Locator, type Page } from "@playwright/test";
import { StaffShellPage } from "../core/StaffShellPage";

export class StaffDashboardPage extends StaffShellPage {
  protected readonly path = "/staff/dashboard";
  readonly overdueInvoicesBody: Locator;
  readonly expiringContractsBody: Locator;
  readonly expiringInvoicesBody: Locator;

  /**
   * Initializes this page object.
   */
  constructor(page: Page) {
    super(page);
    this.overdueInvoicesBody = this.page.locator("#overdueInvoicesBody");
    this.expiringContractsBody = this.page.locator("#expiringContractsBody");
    this.expiringInvoicesBody = this.page.locator("#expiringInvoicesBody");
  }

  async waitForLoaded(): Promise<void> {
    await this.page.waitForURL(/\/staff\/dashboard/);
    await this.waitForTitleContainsLoose("bang dieu khien nhan vien", "dashboard");
    await this.waitForVisible(this.page.locator('a.nav-link.active[href="/staff/dashboard"]'));
  }

  async waitForSummarySectionsVisible(): Promise<void> {
    await this.waitForVisible(this.page.locator("#buildingCntStat"));
    await this.waitForVisible(this.page.locator("#contractCntStat"));
    await this.waitForVisible(this.page.locator("#customerCntStat"));
    await this.waitForVisible(this.page.locator("#unpaidInvoiceCntStat"));
    await this.waitForVisible(this.overdueInvoicesBody);
    await this.waitForVisible(this.expiringContractsBody);
    await this.waitForVisible(this.expiringInvoicesBody);
  }
}

File: automation-test-selenium/src/main/java/org/suntower/pages/staff/StaffDashboardPage.java
package org.suntower.pages.staff;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.suntower.core.BasePage;

public class StaffDashboardPage extends BasePage {
  private final By buildingCount = css("#buildingCntStat");
  private final By contractCount = css("#contractCntStat");
  private final By customerCount = css("#customerCntStat");
  private final By unpaidInvoiceCount = css("#unpaidInvoiceCntStat");
  private final By overdueInvoicesBody = css("#overdueInvoicesBody");
  private final By expiringContractsBody = css("#expiringContractsBody");
  private final By expiringInvoicesBody = css("#expiringInvoicesBody");

  public StaffDashboardPage(WebDriver driver) {
    super(driver);
  }

  public void waitForLoaded() {
    waitForUrlContains("/staff/dashboard");
    waitForVisible(buildingCount);
  }

  public void waitForSummarySectionsVisible() {
    waitForVisible(buildingCount);
    waitForVisible(contractCount);
    waitForVisible(customerCount);
    waitForVisible(unpaidInvoiceCount);
    waitForVisible(overdueInvoicesBody);
    waitForVisible(expiringContractsBody);
    waitForVisible(expiringInvoicesBody);
  }
}
